src/metrics/rules/active_ruleset.go

```diff
@@ -845,6 +856,23 @@ func (as *activeRuleSet) cutoverNanosAt(idx int) int64 {
 	return timeNanosMax
 }
 
+// matchIncludeTag matches the tag name against the list of include tag names and adds to the tag pairs
+// if there is a match. Include tags are tags that if present in the metric id, should be included in the rollup id.
+func matchIncludeTag(
+	tagName []byte,
+	tagVal []byte,
+	includeTagNames [][]byte,
```

Comment:
nit: if includeTagNames is a hashset, then you don't need to do the bytes.Compare as below

Reply:
If we convert includeTagNames to a map, we cannot create a map of []bytes because keys in a map must be comparable in go (https://go.dev/blog/maps). So we would have to create a map of strings as the keys. However, we would then need to convert from []byte to string for each tag name, which could be more time expensive. It may be faster to iterate through the [][]byte.